sources/api/netdog/src/net_config/devices/vlan.rs

```diff
@@ -0,0 +1,87 @@
+use super::{error, Dhcp4ConfigV1, Dhcp6ConfigV1, Result, Validate};
+use super::{validate_addressing, HasDhcp, HasRoutes, HasStatic};
+use crate::interface_name::InterfaceName;
+use crate::net_config::{RouteV1, StaticConfigV1};
+use serde::Deserialize;
+use snafu::ensure;
+use std::convert::TryFrom;
+
+#[derive(Debug, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub(crate) struct NetVlanV1 {
+    pub(crate) primary: Option<bool>,
+    pub(crate) dhcp4: Option<Dhcp4ConfigV1>,
+    pub(crate) dhcp6: Option<Dhcp6ConfigV1>,
+    pub(crate) static4: Option<StaticConfigV1>,
+    pub(crate) static6: Option<StaticConfigV1>,
+    #[serde(rename = "route")]
+    pub(crate) routes: Option<Vec<RouteV1>>,
+    #[allow(dead_code)]
+    kind: VlanKind,
+    pub(crate) device: InterfaceName,
+    pub(crate) id: u16,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(try_from = "String")]
+pub(crate) struct VlanKind(String);
```

Comment:
Can you just use an empty newtype here, and discard the string?
```suggestion
#[serde(try_from = "String")]
pub(crate) struct VlanKind();
```

(same with `BondKind`)

Reply:
I had assumed that we needed to store the string to use `try_from` on String but it looks like you don't. This seems to be what @webern was referring to as well. I'll work on this a bit to refine.